src/treetensornetworks/projttns/abstractprojttn.jl

```diff
@@ -129,13 +129,19 @@ end
 function invalidate_environments(P::AbstractProjTTN)
   ie = internal_edges(P)
   newenvskeys = filter(!in(ie), keys(environments(P)))
-  P = set_environments(P, getindices(environments(P), newenvskeys))
+  T = typeof(environments(P))
```

Comment:
Alternatively, we may want to define a function `safe_getindices` and use it instead of `get_indices`, and reference the issue there?

Reply:
That sounds like a good strategy. We could call it `getindices_narrow_keytype` to be more explicit about what it is doing and how it is different from `getindices`.

Reply:
I define it now in `src/utils.jl`.